server/src/sessionManagement.mjs

```diff
@@ -54,22 +58,259 @@ class Session{
+            }
+            const sessionDataFile = this.sessionOutputPath + `/FireboltCalls_${this.#sessionStart}_${this.sessionOutput}.json`;
+
+            let returnStmt = null; // creating a return message
+            if (this.sessionOutput == 'raw') {
+                console.log("Inside raw format type");
+                fs.writeFileSync(sessionDataFile, sessionDataJson);
+                returnStmt = `Succesfully wrote output in raw format to ${sessionDataFile}`;
+            } else if (this.sessionOutput == "mock-overrides") {
+                console.log("Inside mock-overrides format type");
+                this.convertJsonToYml(sessionDataJson);
+                returnStmt = `Succesfully wrote output in mock-overrides format to ${this.mockOutputPath}`;
+            } else {
+                //emit this json in time order.
+                console.log("Inside log format type");
+                this.sortJsonByTime(sessionData, sessionDataFile);
+                returnStmt = `Succesfully wrote output in log format to ${sessionDataFile}`;
+            }
+
+            return returnStmt;
         } catch (error) {
             logger.error("Error exporting session: " + error);
+            console.log(error.stack);
             return null;
         }
         
     }
 
-    sortJsonByTime(sessionDataJson) {
+    convertJsonToYml(jsonReport) {
+        if (!fs.existsSync(this.mockOutputPath)) {
+            logger.info("Mock directory did not exist direcotry: " + this.mockOutputPath)
+            fs.mkdirSync(this.mockOutputPath, { recursive: true});
+        }
+        let repetition = false;
+        try {
+            jsonReport = JSON.parse(jsonReport);
+        } catch (e) {
+            console.error('Invalid Input format');
+            console.error("Error", e);
+            return e;
+        }
+
+        let methodCallsWritten = []; // creating an array to track which calls have already been written
+        for ( let i = 0; i < jsonReport.calls.length; i++) {
+            // skip methods containing .on & Changed in fullTitle and .Subscribe (SKIPPING THE SUBSCRIBE VALIDATION AS WE ARE NOT CURRENTLY LOGGING) in title
+            if (jsonReport.calls[i].methodCall.indexOf('.on') !== -1 && jsonReport.calls[i].methodCall.indexOf('Changed') !== -1) {
+                continue;
+             } else if (methodCallsWritten.includes(jsonReport.calls[i].methodCall)) { // if already added because of multiple can skip
+                continue;
+             } else {
+                let calls = jsonReport.calls;
+                let multipleExampleFlag = false;
+                let multipleExampleContext = [];
+                let multipleParams;
+                let methodName;
+
+                // skip .Subscribe (SKIPPING THE SUBSCRIBE VALIDATION AS WE ARE NOT CURRENTLY LOGGING) methods
+                for ( let j = i+1; j < calls.length; j++) { // iterate over rest of calls                     
```

Comment:
I don't see us removing calls[j]. If we have 3 calls, wouldn't we get to this point twice? Once when going through the first call and again when going through the second.

Have we tested this scenario?

Reply:
From JB: "Yes this has been tested, so say we have 3 calls with the same method call i.e. advertising.config. If they all have different request parameters than they all should be added to 'multipleExampleContext' to then generate a .yml file. If any of them are the same call meaning they have the same params then the 'checkParams()' will equate to false and the repetition flag will stop a duplicate from being added to 'multipleExampleContext'. So we are making sure we capture any calls that belong together and then not adding them if they are duplicates."